theseus/embodied/robot/joint.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+import warnings
+from typing import Optional
+import torch
+
+from theseus.geometry.functional import so3, se3
+
+
+class Joint(abc.ABC):
+    def __init__(
+        self,
+        name: str,
+        parent: int = -1,
+        child: int = -1,
```

Comment:
These could be `Joint` types instead of `int`. Are there any tradeoffs worth considering?

Reply:
I think these are made `Link` types in a later PR.